packages/ember-data/lib/system/store.js

```diff
@@ -2040,7 +2042,7 @@ function _commit(adapter, store, operation, record) {
 
     store._adapterRun(function() {
       if (adapterPayload) {
-        payload = serializer.extract(store, type, adapterPayload, get(record, 'id'), operation);
+        payload = serializer.extract(store, type, adapterPayload, get(snapshot, 'id'), operation);
```

Comment:
Since `snapshot` is a POJO we're doing `snapshot.id` most places in the codebase.